lib/jxl/test_utils.cc

```diff
@@ -865,6 +865,47 @@ Status EncodeFile(const CompressParams& params, CodecInOut* io,
   return true;
 }
 
+extras::JXLCompressParams CompressParamsForLossless() {
+  extras::JXLCompressParams cparams;
+  cparams.AddOption(JXL_ENC_FRAME_SETTING_MODULAR, 1);
+  cparams.AddOption(JXL_ENC_FRAME_SETTING_COLOR_TRANSFORM, 1);
+  cparams.AddOption(JXL_ENC_FRAME_SETTING_MODULAR_PREDICTOR, 6);  // Weighted
+  cparams.distance = 0.0;
+  return cparams;
+}
+
+StatusOr<ImageF> GetImage(const extras::PackedPixelFile& ppf) {
+  JxlMemoryManager* memory_manager = MemoryManager();
+  ImageF gray;
+  JXL_ENSURE(!ppf.frames.empty());
+  const extras::PackedImage& image = ppf.frames[0].color;
+  const JxlPixelFormat& format = image.format;
```

Comment:
We should add JXL_RETURN_IF_ERROR(format.num_channels == 1), since this function should only be used if we know that the image is grayscale.

Reply:
good idea, done

Reply:
also for the 3 channel case